matrix/tests_matrix.py

```diff
@@ -0,0 +1,181 @@
+import unittest
+
+from matrix import Matrix
+
+
+class TestMatrix(unittest.TestCase):
+    def test_initialization(self):
+        Matrix((1, 2), (3, 4))
+        Matrix((1, 2), (5.5, 4))
+
+        with self.assertRaises(ValueError):
+            Matrix((1, 'a'), (3, 4))
+
+        with self.assertRaises(ValueError):
+            Matrix((1, 2), (3, [4, 5, 6]))
+
+        with self.assertRaises(ValueError):
+            Matrix((1, 2), (3, 4, 5))
+
+    def test_modification(self):
+        m = Matrix((1, 1), (2, 2))
+
+        with self.assertRaises(AttributeError):
+            m._matrix = ((1, 2), (3, 4))
+
+        with self.assertRaises(AttributeError):
+            m._matrix = "hello world!"
+
+    def test_str(self):
+        m = Matrix((1, 1, 1), (2, 2, 2), (3, 3, 3))
+
+        self.assertEqual(str(m), str(((1, 1, 1), (2, 2, 2), (3, 3, 3))))
```

Comment:
Why not save the tuple in variable first?
This way you don't need to duplicate it, and its easy to change if needed